tests/unit/test_analytics.py

```diff
@@ -0,0 +1,163 @@
+    dummy_date = datetime(2024, 5, 2)
+    dummy_datetime_list = [dummy_date + timedelta(seconds=i) for i in range(10)]
+    test_time = pl.Series("time", dummy_datetime_list)
+    window_data = np.array([0, 1, 1, 1, 0, 1, 0, 0, 0, 1])
+    window_measurement = models.Measurement(measurements=window_data, time=test_time)
+    expected_result = [
+        (dummy_datetime_list[1], dummy_datetime_list[4]),
+        (dummy_datetime_list[5], dummy_datetime_list[6]),
+        (dummy_datetime_list[8], dummy_datetime_list[9]),
+    ]
+
+    result = sleep_detection._find_periods(window_measurement)
+
+    assert result == expected_result, f"Expected {expected_result}, but got {result}"
+
+
+def test_spt_window(sleep_detection: analytics.SleepDetection) -> None:
+    """Test the _spt_window method."""
+    sleep_detection.anglez.measurements = np.zeros(
+        len(sleep_detection.anglez.measurements)
+    )
+    half_long_block = 180
+    expected_length = int(len(sleep_detection.anglez.measurements) / 5) - 1
+    expected_result = np.zeros(expected_length)
+    expected_result[half_long_block:] = 1
+
+    result = sleep_detection._spt_window(sleep_detection.anglez)
+
+    assert np.array_equal(
+        result.measurements, expected_result
+    ), f"Expected {expected_result}, but got {result.measurements}"
+    assert np.array_equal(
+        len(result.time), expected_length
+    ), f"Expected {expected_length}, but got {len(result.time)}"
+
+
+def test_calculate_sib_periods(sleep_detection: analytics.SleepDetection) -> None:
+    """Test the _calculate_sib_periods method."""
+    expected_length = math.ceil(len(sleep_detection.anglez.measurements) / 300)
+    expected_result = np.zeros(expected_length)
+
+    result = sleep_detection._calculate_sib_periods(sleep_detection.anglez, 10)
+
+    assert np.array_equal(
+        result.measurements, expected_result
+    ), f"Expected {expected_result}, but got {result.measurements}"
+    assert (
+        len(result.measurements) == expected_length
+    ), f"Expected {expected_length}, but got {len(result.measurements)}"
+
+
+def test_find_onset_wakeup_times(sleep_detection: analytics.SleepDetection) -> None:
+    """Test the _find_onset_wakeup_times method."""
+    dummy_date = datetime(2024, 5, 2)
+    spt_periods = [
+        (
+            dummy_date + timedelta(hours=1),
+            dummy_date + timedelta(hours=3),
+        )
+    ]
+    sib_periods = [
+        (
+            dummy_date + timedelta(hours=2),
+            dummy_date + timedelta(hours=4),
+        )
+    ]
+    expected_output = analytics.SleepWindow(
+        onset=[dummy_date + timedelta(hours=2)],
+        wakeup=[dummy_date + timedelta(hours=4)],
+    )
+
+    result = sleep_detection._find_onset_wakeup_times(spt_periods, sib_periods)
+
+    assert result.onset == expected_output.onset
+    assert result.wakeup == expected_output.wakeup
+
+
+def test_run_sleep_detection(sleep_detection: analytics.SleepDetection) -> None:
```

Comment:
Think i forgot to mention this in the last meeting, but I'll write this again, run doesn't actually do anything but call all the other functions in the right order, so really what kind of edge cases could there be?

Also eventually run will do one additional thing which is to pick the right method,  should we write the test for that now so we have it as this expands. @ReinderVosDeWael

Reply:
Function has changed with new `GGIRSleepDetector`. I guess I'm just testing that we get the correct result when running the complete `sleep_detection.run_sleep_detection()`